viewing_party/party.py

```diff
@@ -1,23 +1,121 @@
 # ------------- WAVE 1 --------------------
 
+from shutil import move
+
+
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        return {
+            "title": title,
+            "genre": genre,
+            "rating": rating,
+        }
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data.setdefault("watched", []).append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data.setdefault("watchlist", []).append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:

```

Comment:
Note, while this may work here, it's considered dangerous practice to modify the content of a data structure that you're looping over.  Especially in more complex code, this can open you up to complex errors. A simple alternative here could be to save a copy of what you get from movie in user_data["watchlist"] and iterate over that, so that as you modify user_data["watchlist"], there's no risk.

Reply:
I think it would be helpful for me to talk through what this alternative would look like in code. Just flagging here that I'll probably swing by your office hours next week to ask about this!